### PR TITLE
Fix for issue #319

### PR DESCRIPTION
**Pull request by TheTrueCaligari**
I tried to fix the Upcoming/Due transactions not being summed in the *Selected Balance*.

The problem seems to come from the fact that those transactions are *not* transactions, hence they don't appear in the sum query.
I think I fixed it with a loop on the different schedules, taking only the right ones (and fixing the sign for transfers with the additional parameter *account*).

I also tried to do the same thing using another sum query, something like this:

```javascript
  let scheduleBalance = useSheetValue({
    name: name + '-schedule-sum',
    query: q('schedules')
      .filter({ id: { $oneof: previewIds } })
      .options({ splits: 'all' })
      .calculate({ $sum: '$_amount' }),
  });
```

but it didn't work :(

### CODE REVIEW

File: packages/desktop-client/src/components/accounts/Account.js [hunk 1]
```diff
@@ -369,9 +369,22 @@ function SelectedBalance({ selectedItems }) {
...
+    if (previewIds.includes(s.id))
+      if (!account || account.id === s._account) scheduleBalance += s._amount;
+      else scheduleBalance -= s._amount;
   }
+
+  if (balance == null) {
+    if (scheduleBalance == null) return null;
+    else balance = scheduleBalance;
+  } else if (scheduleBalance != null) balance += scheduleBalance;
+
```
Comment (j-f1) [inline packages/desktop-client/src/components/accounts/Account.js:387]: Can you add braces to all of these if statements? I find it a bit hard to read this way. (Also opened #1015 to enforce this more often in the future)

**Review by j-f1 — COMMENTED** [review]
Functionality looks good to me! One small style nit:

File: packages/desktop-client/src/components/accounts/Account.js (hunk 1 above)
Comment (TheTrueCaligari) [reply packages/desktop-client/src/components/accounts/Account.js]: Sure thing !

I wasn't too fine with the original presentation either, I was just trying to comply to the rules of eslint.
I hope things looks ok now :)

**Review by j-f1 — APPROVED** [review]
LGTM!

**Pull request merged**